src/React.re

```diff
@@ -139,10 +456,428 @@ module Suspense = {
+  unit =
+  "useInsertionEffect";
+[@mel.module "react"]
+external useInsertionEffect6:
+  (
+    [@mel.uncurry] (unit => option(unit => unit)),
+    ('a, 'b, 'c, 'd, 'e, 'f)
+  ) =>
+  unit =
+  "useInsertionEffect";
+[@mel.module "react"]
+external useInsertionEffect7:
+  (
+    [@mel.uncurry] (unit => option(unit => unit)),
+    ('a, 'b, 'c, 'd, 'e, 'f, 'g)
+  ) =>
+  unit =
+  "useInsertionEffect";
+
+[@mel.module "react"]
+external useLayoutEffect:
+  ([@mel.uncurry] (unit => option(unit => unit))) => unit =
+  "useLayoutEffect";
+[@mel.module "react"]
+external useLayoutEffect0:
+  (
+    [@mel.uncurry] (unit => option(unit => unit)),
+    [@mel.as {json|[]|json}] _
+  ) =>
+  unit =
+  "useLayoutEffect";
+[@mel.module "react"]
+external useLayoutEffect1:
+  ([@mel.uncurry] (unit => option(unit => unit)), array('a)) => unit =
+  "useLayoutEffect";
+[@mel.module "react"]
+external useLayoutEffect2:
+  ([@mel.uncurry] (unit => option(unit => unit)), ('a, 'b)) => unit =
+  "useLayoutEffect";
+[@mel.module "react"]
+external useLayoutEffect3:
+  ([@mel.uncurry] (unit => option(unit => unit)), ('a, 'b, 'c)) => unit =
+  "useLayoutEffect";
+[@mel.module "react"]
+external useLayoutEffect4:
+  ([@mel.uncurry] (unit => option(unit => unit)), ('a, 'b, 'c, 'd)) => unit =
+  "useLayoutEffect";
+[@mel.module "react"]
+external useLayoutEffect5:
+  ([@mel.uncurry] (unit => option(unit => unit)), ('a, 'b, 'c, 'd, 'e)) =>
+  unit =
+  "useLayoutEffect";
+[@mel.module "react"]
+external useLayoutEffect6:
+  (
+    [@mel.uncurry] (unit => option(unit => unit)),
+    ('a, 'b, 'c, 'd, 'e, 'f)
+  ) =>
+  unit =
+  "useLayoutEffect";
+[@mel.module "react"]
+external useLayoutEffect7:
+  (
+    [@mel.uncurry] (unit => option(unit => unit)),
+    ('a, 'b, 'c, 'd, 'e, 'f, 'g)
+  ) =>
+  unit =
+  "useLayoutEffect";
+
+[@mel.module "react"]
+external useMemo: ([@mel.uncurry] (unit => 'any)) => 'any = "useMemo";
+[@mel.module "react"]
+external useMemo0:
+  ([@mel.uncurry] (unit => 'any), [@mel.as {json|[]|json}] _) => 'any =
+  "useMemo";
+[@mel.module "react"]
+external useMemo1: ([@mel.uncurry] (unit => 'any), array('a)) => 'any =
+  "useMemo";
+[@mel.module "react"]
+external useMemo2: ([@mel.uncurry] (unit => 'any), ('a, 'b)) => 'any =
+  "useMemo";
+[@mel.module "react"]
+external useMemo3: ([@mel.uncurry] (unit => 'any), ('a, 'b, 'c)) => 'any =
+  "useMemo";
+[@mel.module "react"]
+external useMemo4: ([@mel.uncurry] (unit => 'any), ('a, 'b, 'c, 'd)) => 'any =
+  "useMemo";
+[@mel.module "react"]
+external useMemo5:
+  ([@mel.uncurry] (unit => 'any), ('a, 'b, 'c, 'd, 'e)) => 'any =
+  "useMemo";
+[@mel.module "react"]
+external useMemo6:
+  ([@mel.uncurry] (unit => 'any), ('a, 'b, 'c, 'd, 'e, 'f)) => 'any =
+  "useMemo";
+[@mel.module "react"]
+external useMemo7:
+  ([@mel.uncurry] (unit => 'any), ('a, 'b, 'c, 'd, 'e, 'f, 'g)) => 'any =
+  "useMemo";
+
+/* This is used as return values  */
```

Comment:
This was for `type callback('input, 'output)`. which is no longer here

```suggestion
/* This is used as return values  */
```

Reply:
Removed this and added into the React.Uncurried callback